### PR TITLE
Wasm lib first testing

### PR DESCRIPTION
**Pull request by barbariand**
adding the start for wasm32-unknown-unknown support

### CODE REVIEW

**Comment by Alvinn8** [comment]
Maybe the directory should be lowercase?

**Comment by barbariand** [comment]
Yea you'r right, i will fix it

File: .gitignore [hunk 1]
```diff
@@ -12,10 +12,10 @@ Cargo.lock
...
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
-
 .idea
 .vscode
 
 # Added by cargo
 
 /target
+/Reti-js/wasm/**
```
Comment (Alvinn8) [inline .gitignore]:
```suggestion
/reti-js/wasm/**
```

File: reti-js/test.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+
+</head>
+
+<body>
+    <script type="module">
+        import init, { parse } from "./wasm/Reti_js.js"
```
Comment (Alvinn8) [inline reti-js/test.html]:
```suggestion
        import init, { parse } from "./wasm/reti_js.js"
```

File: reti-js/init_test.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin bash
+# This is a script for running the building and setup so you can use it from the index.html file exported as the window.test function
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+dir=$(pwd);
+cd $SCRIPT_DIR
+echo "building"
+FLAGS="-C opt-level=z -C lto";
+env RUSTFLAGS="$FLAGS" sh -c 'echo "running with flags $RUSTFLAGS" & cargo build --release --target wasm32-unknown-unknown'
+echo "binding wasm"
+wasm-bindgen --web $SCRIPT_DIR/../target/wasm32-unknown-unknown/release/Reti_js.wasm --out-dir $SCRIPT_DIR/wasm
```
Comment (Alvinn8) [inline reti-js/init_test.sh]:
```suggestion
wasm-bindgen --web $SCRIPT_DIR/../target/wasm32-unknown-unknown/release/reti_js.wasm --out-dir $SCRIPT_DIR/wasm
```

**Review by Alvinn8 — APPROVED** [review]
Very cool! Just some small changes to lowercase stuff.

**Pull request merged**